test/gtest/activation.cpp

```diff
@@ -0,0 +1,243 @@
+
+        CompareTensors(dinput_cpu, dinput_gpu);
+        miopenDestroyActivationDescriptor(activ_desc);
+    }
+
+    tensor<float> input; // x
+    tensor<float> output_gpu;
+    tensor<float> output_cpu_ref; // y
+
+    tensor<float> dinput_cpu; // dx
+    tensor<float> dinput_gpu;
+    tensor<float> doutput;
+
+    ActivationConfig activ_config;
+    miopenActivationDescriptor_t activ_desc;
+    miopen::Allocator::ManageDataPtr in_dev;   // x
+    miopen::Allocator::ManageDataPtr out_dev;  // y
+    miopen::Allocator::ManageDataPtr din_dev;  // dx
+    miopen::Allocator::ManageDataPtr dout_dev; // dy
+};
+
+miopenStatus_t RunActivation(miopen::Handle& handle,
+                             miopenActivationDescriptor_t activationDesc,
+                             const void* alpha,
+                             const miopen::TensorDescriptor& xDesc,
+                             ConstData_t x,
+                             const void* beta,
+                             const miopen::TensorDescriptor& yDesc,
+                             Data_t y,
+                             const miopen::TensorDescriptor& dyDesc,
+                             ConstData_t dy,
+                             const miopen::TensorDescriptor& dxDesc,
+                             Data_t dx)
+{
+    if(alpha == nullptr || beta == nullptr)
+        MIOPEN_THROW(miopenStatusBadParm, "alpha or beta is NULL");
+
+    miopen::ActivationDescriptor desc = miopen::deref(activationDesc);
+    miopenStatus_t fwdStatus          = desc.Forward(handle,
+                                            alpha,
+                                            xDesc, // input.desc
+                                            x,     // in_dev.get()
+                                            beta,
+                                            yDesc, // output_gpu.desc
+                                            y);    // out_dev.get()
+
+    miopenStatus_t bwdStatus = desc.Backward(handle,
+                                             alpha,
+                                             yDesc, // out.desc
+                                             y,     // out_dev.get()
+                                             dyDesc,
+                                             dy,
+                                             xDesc, // input.desc
+                                             x,
+                                             beta,
+                                             dxDesc,
+                                             dx);
+
+    if(fwdStatus == miopenStatusSuccess && bwdStatus == miopenStatusSuccess)
+        return miopenStatusSuccess;
+    else if(fwdStatus != miopenStatusSuccess)
+        return fwdStatus;
+    else
+        return bwdStatus;
+}
+
+INSTANTIATE_TEST_SUITE_P(ActivationTestSuite,
+                         TestActivation,
+                         ::testing::Values(ActivationConfig{16, 32, 8, 8, miopenActivationELU}));
```

Comment:
Please add more test cases to cover 
* A few other sizes
* All supported activations. 
One test vector seems to be under-whelming.

Reply:
> Please add more test cases to cover
> 
>     * A few other sizes
> 
>     * All supported activations.
>       One test vector seems to be under-whelming.

Updated with 7 cases now and 9 activations. The following haven't passed though, miopenActivationLOGISTIC, miopenActivationTANH, miopenActivationPOWER, miopenActivationCLIPPEDRELU